src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -2773,15 +2773,23 @@ private TypeSyntax ParseReturnType()
 
         private bool IsEndOfReturnType()
         {
-            switch (this.CurrentToken.Kind)
+            if (this.CurrentToken.Kind == SyntaxKind.IdentifierToken)
             {
-                case SyntaxKind.OpenParenToken:
-                case SyntaxKind.OpenBraceToken:
-                case SyntaxKind.SemicolonToken:
-                    return true;
-                default:
-                    return false;
+                switch (this.PeekToken(1).Kind)
+                {
+                    case SyntaxKind.OpenParenToken: // methodName (
+                    case SyntaxKind.OpenBraceToken: // propertyName {
+                    case SyntaxKind.SemicolonToken: // fieldName ;
```

Comment:
what about .EqualsToken for "fieldName ="?

Reply:
also, ```MethodName<```.

Note: if these shoudl *not* be checked because they could be part of the return type, then we should document that here as intentional omissions.